interfaces/Portalicious/src/app/services/paginate-query.service.ts

```diff
@@ -248,4 +249,20 @@ export class PaginateQueryService {
       previewItem: selection[0],
     };
   }
+
+  public getActionDataForRegistration(
+    registration: Registration,
+  ): ActionDataWithPaginateQuery<Registration> {
+    return {
+      query: {
+        filter: {
+          referenceId: `${FilterOperator.EQ}:${registration.referenceId}`,
+        },
+      },
+      count: 1,
+      selection: [],
+      selectAll: false,
+      previewItem: registration,
+    };
+  }
```

Comment:
I do think this could be made to be generic, and to reuse the function we already have.

```suggestion
  public singleItemToActionData<TData>({
    item,
    fieldForFilter,
  }: {
    item: TData;
    fieldForFilter: keyof TData & string;
  }) {
    return this.selectionEventToActionData({
      selection: [item],
      fieldForFilter,
      totalCount: 1,
      previewItemForSelectAll: item,
    });
  }
```

Reply:
>  I don't see such a pattern very often in Portalicious. Is there are reason for that?

Not a conscious one. I suppose we just have done more "guarding" rather than "throwing".

That said, I think you can remove the `| undefined` in the return type of `selectionEventToActionData`, and it should solve this problem. Tbh I don't know why it's there to begin with.